### ISSUE TITLE
ICU 72 doesn't include sv_reformed_meta.toml

### ISSUE THREAD

**Issue by sffc** [unicode-org/icu4x#2764]
Running datagen against the new ICU 72 zip file prints out:

```
WARN  [icu_provider::error] ICU4X data error: Zip: specified file not found in archive
WARN  [icu_provider::error] ICU4X data error: Zip: collation/implicithan/sv_reformed_meta.toml
```

What does it mean? Does something need to be fixed?

CC @echeran @markusicu @hsivonen

**Comment by echeran** [comment]
The file `provider/datagen/src/transform/icuexport/collator/mod.rs` refers to [CLDR-15603](https://unicode-org.atlassian.net/browse/CLDR-15603), which requests the naming for the `sv` locale collation data in CLDR, which is reflected in the file names in `icuexportdata` files, be consistent with the naming convention for all other locales.  So the changes made were "standard" -> "traditional", and "reformed" -> "standard".

The `icuexportdata` bundle contains the following files matching `/collation/implicithan/sv*`:
```
sv_search_data.toml
sv_search_meta.toml
sv_standard_data.toml
sv_standard_meta.toml
sv_traditional_data.toml
sv_traditional_meta.toml
```

I think we want to update the entries in `provider/testdata/Cargo.toml` accordingly, and remove `reformed_swedish_exists()` from the `.../icuexport/collator/mod.rs` file mentioned above.

@hsivonen Does that sound right?

**Comment by hsivonen** [comment]
It's expected that "reformed" is gone from the new export zip. I think `reformed_swedish_exists()` should stay for now in order to keep ICU4X able to consume both ICU 71 and 72 exports. Is it not working gracefully with the 72 export? `reformed_swedish_exists()` seemed to work with 72 pre-release data when I added `reformed_swedish_exists()`.

**Comment by echeran** [comment]
I created #2770 to represent the future work to update ICU4X to use Unicode 15.1 data from ICU 72 `icuexportdata`, and I'll mark this issue here as blocked on that new one.

**Comment by sffc** [comment]
Discussion:

- @echeran's PR is what we want for ICU4X 2.0, but we can't merge it now because we want to maintain CLDR 41 support
- @robertbastian knows how to fix this warning
